### PR TITLE
Edition update

### PR DESCRIPTION
**Pull request by mthebridge**
Update to 2018 edition, and take new version of slog to allow new-style macro imports.

### CODE REVIEW

File: slog-extlog-derive/src/lib.rs [hunk 1]
```diff
@@ -173,14 +156,9 @@
 
 #![recursion_limit = "128"]
 
-//#![feature(trace_macros)]
-//trace_macros!(true);
-
 extern crate proc_macro;
 #[macro_use]
 extern crate quote;
```
Comment (tarquin-the-brave) [inline slog-extlog-derive/src/lib.rs:161]: still some `extern crate`s here.

File: slog-extlog-derive/src/lib.rs [hunk 2]
```diff
@@ -560,35 +538,35 @@ fn impl_loggable(ast: &syn::DeriveInput) -> quote::Tokens {
     // Generate the actual log call based on the provided level.
     let match_gen = match level {
         Level::Critical => {
-            quote! { crit!(logger, #text; "log_id" => id_val, #(#fields, )* "details" => self) }
+            quote! { slog::crit!(logger, #text; "log_id" => id_val, #(#fields, )* "details" => self) }
```
Comment (tarquin-the-brave) [inline slog-extlog-derive/src/lib.rs:541]: at the top, would it be better to put a:
```rust
use slog::{debug, info, warn, crit, error, trace}
```
and some other `slog` things used in this file?

**Review by tarquin-the-brave — COMMENTED** [review]
Looks good.  I've ran `fmt` and `clippy` over it and it's all sound.
A couple of comment about left over `extern crate`s and places where sticking a `use` at the top could be handy.

File: slog-extlog-derive/src/lib.rs (hunk 1 above)
Comment (mthebridge) [reply slog-extlog-derive/src/lib.rs:161]: Yeah, that was deliberate.
- `proc_macro` is compiler magic that can;t be removed.
- `quote` isn't 2018-ified so you have to import all the dependenct macros as well, many of which are meant to be private.  Seemed easier to leave in the `macro_use`.

File: slog-extlog-derive/src/lib.rs (hunk 2 above)
Comment (mthebridge) [reply slog-extlog-derive/src/lib.rs:541]: As we discussed - this `quote!` macro is building up the egenrated code for a `derive` impl - the `use` would need to be in the code that has `#[derive(ExtLoggable)]`.  It's much better to not require the user to `use` other macros in order to make your custom derive work.

File: slog-extlog-derive/src/lib.rs (hunk 1 above)
Comment (tarquin-the-brave) [reply slog-extlog-derive/src/lib.rs:161]: fine.

**Pull request merged**